karabo_bridge/simulation.py

```diff
@@ -185,180 +177,31 @@ class LPD(Detector):
     ])
 
 
-def generate(detector, nsources):
-    tid_counter = 10000000000
-    while True:
-        data, meta = detector.gen_data(tid_counter)
+class DataGenerator:
```

Comment:
Does converting this to a class actually gain us anything? I don't have any objection to it, I just think generator functions are quite a nice, concise way to write code.

Reply:
I don't remember exactly why, was part of a different route I did not follow... changed it back to a function